src/codelens/bazel_build_code_lens_provider.ts

```diff
@@ -20,6 +21,48 @@ import { getDefaultBazelExecutablePath } from "../extension/configuration";
 import { blaze_query } from "../protos";
 import { CodeLensCommandAdapter } from "./code_lens_command_adapter";
 
+/** Returns a target string without some prefix.
+ *
+ * If the target string does not have the prefix, returns the string unchanged.
+ *
+ * @param target The string whose prefix to remove.
+ * @param prefix The prefix to remove, if present.
+ * @returns The target string without the provided prefix.
+ */
+function withoutPrefix(target: string, prefix: string): string {
+  if (!target.startsWith(prefix)) {
+    return target;
+  }
+  return target.substring(prefix.length);
+}
+
+/** Computes the shortened name of a Bazel target.
+ *
+ * For example, if the target name starts with `//foo/bar/baz:fizbuzz` and we're
+ * in `foo/bar/baz/BUILD.bazel`, the target's short name will be `fizzbuzz`.
+ *
+ * This allows our code lens suggestions to avoid filling users' screen with
+ * redundant path information.
+ *
+ * @param targetName The unshortened name of the target.
+ * @param bazelWorkspaceInfo The currrent Bazel workspace.
+ * @param location The location of the target's BUILD.bazel file.
+ *
+ * @returns The shortened name of the target.
+ */
+function getTargetShortName(
```

Comment:
Can't we always strip the package here, since targets we're considering will always be in the current package?

Reply:
I wasn't sure if this was true (whether target rule names are always within the current package, or whether it's possible to declare something outside of the current package). I can't think of a way you'd break this, though. Changed.